lib/Logger.js

```diff
@@ -11,46 +11,22 @@ class Logger {
      * This is the constructor of the Logger class. It takes optional
      * configuration parameters, that allow to modify its behavior.
      *
+     * @param {Werelogs.Config} config  - An instanciated Werelogs Config object
+     *
      * @param {string} name  - The name of the Logger. It can be found later on
      *                         in the log entries.
      *
-     * @param {object} config         - A configuration object for werelogs.
-     * @param {string} config.level   - The string name of the logging level
-     *                                  ('trace', 'debug', 'info', 'warn',
-     *                                  'error' and 'fatal' in order of
-     *                                  importance.)
-     * @param {string} config.dump    - The string name of the log dumping level
-     *                                  ('trace', 'debug', 'info', 'warn',
-     *                                  'error' and 'fatal' in order of
-     *                                  importance.)
-     * @param {object[]} config.streams   - The array of streams into which to
-     *                                      log. Their configuration is directly
-     *                                      related to the expected bunyan
-     *                                      streams array, for compatibility
-     *                                      purposes (except that the 'level'
-     *                                      field is not accounted for)
-     *
-     * @see [Bunyan's documentation]{@link
-     * https://github.com/trentm/node-bunyan/blob/master/README.md#streams} for
-     * a more detailed description of the streams array configuration.
-     *
      * @returns {undefined}
      */
-    constructor(name, config) {
+    constructor(config, name) {
+        if (!(config instanceof Config)) {
+            throw new TypeError('Invalid parameter Type for "config".');
+        }
+        if (!(typeof name === 'string' || name instanceof String)) {
```

Comment:
This is handled by Config, when we instanciate a Config object, or update it.

As such, this is no longer useful to do this check in Logger.js, as the responsibility moved around.